cw/cw305/simple_capture_traces_batch.py

```diff
@@ -79,6 +81,16 @@ def run_batch_capture(capture_cfg, ot, ktp, scope):
             # Transfer traces
             waves = scope.capture_and_transfer_waves()
             assert waves.shape[0] == scope.num_segments
+            # Check that the recorded traces are within [1, 2**resolution-2] to ensure the ADC
+            # doesn't saturate.
+            if not (np.all(np.greater(waves, adc_range[0]))
```

Comment:
nit: Since there is some duplication of these routines. It may be worth to move this inside device.py or some utility module.

Reply:
Good point. I've now added a function `check_range()` that can be called from both capture scripts.